src/main/kotlin/com/example/olaf/blogtutorial/HttpControllers.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.olaf.blogtutorial
+
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.server.ResponseStatusException
+
+@RestController
+@RequestMapping("/api/article")
+class ArticleController(private val repository: ArticleRepository) {
+    @GetMapping("/")
+    fun findAll() = repository.findAllByOrderByAddedAtDesc()
+
+    @GetMapping("/{slug}")
+    fun findOne(@PathVariable slug: String) =
+        repository.findBySlug(slug) ?: ResponseStatusException(HttpStatus.NOT_FOUND, "This article does not exist!!")
```

Comment:
```suggestion
        repository.findBySlug(slug) ?: ResponseStatusException(HttpStatus.NO_CONTENT, "This article does not exist!!")
```
404 는 uri 자체가 없다는 뜻인데 이건 조회할 게 없는 것이라서
엄밀히? 보면 저는 404는 아닌것 같고
그렇다면 뭘 줄까 하다가 제일 적당한게 [204 not content](https://tools.ietf.org/html/rfc7231#section-6.3.5) 라고 생각하고
이건 2xx 라서 성공은 성공인데, 뭐가 없다 뭐 그런 뜻...

뭐 이번 피알에서 중한건 아니니 그냥 생각만 해보시고 넘어가시면 좋을 듯 합니다.

Reply:
생각해보니까 성공은 맞는데 없다는게 더 맞는 말인 것 같네요 감사합니다 👍